### PR TITLE
Make DataFrame.to_html output full content

### LINKED ISSUE #17004: Make DataFrame.to_html output full content
Make DataFrame.to_html output full contents by default. Or at least add option to enable outputting full content. 

#### Code Sample
```python
import pandas as pd
df = pd.DataFrame({
    'id': [1, 2, 3, 4, 5],
    'link': ['http://a_super_loooooooooooooooooooooooooooooooong_link.com', '', '', '', '']
})
df.to_html('df.html')
```
#### Problem description

By default, pandas uses a short representation of a very long string. However, when outputting to HTML, I think users expect the full content to be outputted rather than a short representation. 

#### Expected Output

Full content output is expected. Or least there should be an option to enable outputting full content.

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ ] closes #17004
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/formats/test_to_html.py
Comment (jreback) [inline pandas/tests/io/formats/test_to_html.py]: can you be specificerror when using noqa

File: pandas/io/formats/html.py
Comment (jreback) [inline pandas/io/formats/html.py]: this is an api change yes?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/html.py
Comment (simonjayhawkins) [reply pandas/io/formats/html.py]: not at the moment, but probably needs to be if we want to have the option to truncate html output.

it is a change of behavior though. whether the current behavior is considered a bug or not.

Comment (jreback) [reply pandas/io/formats/html.py]: sorry, that's what I meant, this will *not* truncate by-default (but it IS truncating now)

File: pandas/tests/io/formats/test_to_html.py
Comment (simonjayhawkins) [reply pandas/tests/io/formats/test_to_html.py]: if your happy using that text, i'll break the string up over several lines instead of using the # noqa directive

Comment (jreback) [reply pandas/tests/io/formats/test_to_html.py]: yes that would be better actually

File: pandas/io/formats/html.py
Comment (simonjayhawkins) [reply pandas/io/formats/html.py]: yes for to_html() only.

the behavior is unchanged for _repr_html_() (a.k.a. to_html(notebook=True) see #23820

File: pandas/tests/io/formats/test_to_html.py
Comment (simonjayhawkins) [reply pandas/tests/io/formats/test_to_html.py]: or make it a fixture, or use an existing fixture. i'm open to suggestions

Comment (jreback) [reply pandas/tests/io/formats/test_to_html.py]: oh a fixture would be good

Comment (simonjayhawkins) [reply pandas/tests/io/formats/test_to_html.py]: top level conftest.py? session scope is ok for immutable string

File: pandas/io/formats/html.py
Comment (simonjayhawkins) [reply pandas/io/formats/html.py]: i was thinking that a change to the api should be part of this PR see https://github.com/pandas-dev/pandas/issues/17004#issuecomment-455821028.

but if we consider `display.max_colwidth` as a _display_ option and `to_html()` as _output_ then i think it is safe to consider #17004 as a bug and leave any api changes out of this PR.

this makes sense considering https://github.com/pandas-dev/pandas/pull/24840#issuecomment-455825159 regarding RC and api changes

File: pandas/conftest.py
Comment (WillAyd) [inline pandas/conftest.py]: Can just define this in the function

File: pandas/conftest.py
Comment (WillAyd) [inline pandas/conftest.py]: Do you need to explicitly state scope? Assumed this to be session given location but maybe have that wrong

File: pandas/conftest.py
Comment (TomAugspurger) [reply pandas/conftest.py]: Or just declare it in test_to_html. That seems easier to follow.

File: pandas/io/formats/html.py
Comment (TomAugspurger) [reply pandas/io/formats/html.py]: Should we wait for 0.24.1 on this?

File: pandas/conftest.py
Comment (simonjayhawkins) [reply pandas/conftest.py]: the scope parameter of the fixture is the scope for which this fixture is shared. the default is to execute the fixture function and all of the dependent fixtures for each parametrized run of each test.
the session scope here just means that the fixture function is only executed once and reused each time. here the fixture setup is trivial so specifying the scope probably doesn't achieve anything.

File: pandas/io/formats/html.py
Comment (simonjayhawkins) [reply pandas/io/formats/html.py]: considering `display.max_rows` and `display.max_columns` are not used for `to_html(max_rows=None, max_cols=None)` then i see no reason why using the`display.max_colwidth` in `to_html()` as a default should not be labelled as a bug and rectified as soon as.

changing the api to _allow_ `display.max_colwidth` to be used in `to_html()` should be considered an enhancement and can wait for a later release.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/html.py
Comment (simonjayhawkins) [reply pandas/io/formats/html.py]: > this _does_ respect the max_colwith now, this is an API change here

@jreback  i'm confused! by _now_ do you mean in the master _now_ or with the changes _now_?

`display.max_rows` and `display.max_columns` are not used as default values in `to_html()`, so why should `display.max_colwidth` be used by default?

it _is_ a change of behaviour, granted. it is a bug fix. it is not a change to the api.

Comment (simonjayhawkins) [reply pandas/io/formats/html.py]: yes the `test_ignore_display_max_colwidth` test will fail. the `test_display_max_colwidth` test is a regression test to check that the notebook display is unaffected by the change. this test passes with and without the code changes.

Comment (TomAugspurger) [reply pandas/io/formats/html.py]: While this is a bug fix, it'll change the behavior, in a way that I think is possibly to large for the RC.

I'd recommend waiting for 0.24.1.

Comment (simonjayhawkins) [reply pandas/io/formats/html.py]: @TomAugspurger 

> I'd recommend waiting for 0.24.1

OK. so what's involved. just leave the PR open?

Comment (simonjayhawkins) [reply pandas/io/formats/html.py]: @TomAugspurger if you want to include this in 0.24.1, is there anything else for me to do here? or would it be best to flag this as 0.25?

Comment (TomAugspurger) [reply pandas/io/formats/html.py]: Let's see what happens to https://github.com/pandas-dev/pandas/issues/24949.

I'm hopefully doing a 0.24.1 in the next day or two that's essentially just regression fixes. If we go with that, then this would be appropriate for 0.24.2 (sorry for bumping it repeatedly).

We don't have a 0.24.2.rst yet, but I'll make one if we plan to do a 0.24.1 release soon.

Comment (jreback) [reply pandas/io/formats/html.py]: if its not an api change, why wouldn't this go in 0.24.1 then?

Comment (simonjayhawkins) [reply pandas/io/formats/html.py]: > 0.24.2 is the _same_ as 0.24.1 in that its meant for bug fixes. If its an api change then its 0.25, which is what I marked, can you move the whatsnew note.

@jreback: sorry that i'm creating unnecessary confusion. my understanding is that 0.24.1 is for regressions only. this PR is a bug fix and being concerned about potentially introducing a new regression, would prefer to see it go in 0.24.2.

The mention of 0.25 is because the fix could be seen as an unexpected change for a users perspective. This PR is not an api change.

**Comment by simonjayhawkins** [comment]
OK. so considering that this bug has such a simple workaround that it is unlikely to be a real issue to most users then i think the best way forward here is to _also_ include an api change in _this_ PR and defer to 0.25.

if I add a `max_colwidth` parameter to `to_html` that just takes an non-negative int (to be consistent with the `max_cols` and `max_rows` parameters of `to_html`), users prefering the current (incorrect) behaviour could then just use `to_html(max_colwidth=pd.options.display.max_colwidth)`

i think issue #17004 could then be considered fully closed.

File: pandas/io/formats/html.py
Comment (WillAyd) [inline pandas/io/formats/html.py]: Can just use `None` here

**Review by WillAyd — CHANGES_REQUESTED** [review]
Minor comment otherwise lgtm

File: pandas/io/formats/html.py
Comment (simonjayhawkins) [reply pandas/io/formats/html.py]: would seem logical, but raises:

`ValueError: Value must have type '<class 'int'>'`

there is a bug/feature where values less than 4 are ignored, see #16097 but that would be more of a hack than using 999999, though would work with strings of length greater than 999999 chars 😀

**Review by WillAyd — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**
